code/site/components/com_pages/dispatcher/router/router.php

```diff
@@ -17,23 +25,97 @@ public function __construct(KObjectConfig $config)
         $this->getObject('manager')->registerAlias($this->getIdentifier(), 'router');
     }
 
-    protected function _initialize(KObjectConfig $config)
+    /**
+     * Resolve a route
+     *
+     * Proxy to the specific component router
+     *
+     * @param string|ComPagesDispatcherRouterRouteInterface|KObjectInterface $route The route to resolve
+     * @param array $parameters Route parameters
+     * @return false| ComPagesDispatcherRouterInterface Returns the matched route or false if no match was found
+     */
+    public function resolve($route, array $parameters = array())
     {
-        $config->append(array(
-            'resolvers'  => array('site', 'page'),
-        ));
+        $result = false;
+
+        if($route instanceof KObjectInterface)
+        {
+            if($route instanceof ComPagesDispatcherRouterRouteInterface) {
+                $package = $route->getScheme();
+            } else {
+                $package = $route->getIdentifier()->getPackage();
+            }
+        }
+        else $package = parse_url($route, PHP_URL_SCHEME);
 
-        parent::_initialize($config);
+        $identifier = 'com://site/'.$package.'.dispatcher.router.'.$package;
+
+        if($resolver = $this->getResolver($identifier)) {
+            $result = $resolver->resolve($route, $parameters);
+        }
+
+        return $result;
     }
 
-    public function getPage($content = false)
+    /**
+     * Generate a route
+     *
+     *  Proxy to the specific component router
+     *
+     * @param string|ComPagesDispatcherRouterRouteInterface|KObjectInterface $route The route to resolve
+     * @param array $parameters Route parameters
+     * @return false|KHttpUrlInterface Returns the generated route
+     */
+    public function generate($route, array $parameters = array())
     {
-        $page = false;
+        $result = false;
+
+        if($route instanceof KObjectInterface)
+        {
+            if($route instanceof ComPagesDispatcherRouterRouteInterface) {
+                $package = $route->getScheme();
+            } else {
+                $package = $route->getIdentifier()->getPackage();
+            }
+        }
+        else $package = parse_url($route, PHP_URL_SCHEME);
+
+        $identifier = 'com://site/'.$package.'.dispatcher.router.'.$package;
+
+        if($resolver = $this->getResolver($identifier)) {
+            $result = $resolver->generate($route, $parameters);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get the resolver
+     *
+     * @param   mixed   $resolver  KObjectIdentifier object or valid identifier string
+     * @return ComPagesDispatcherRouterInterface
+     */
+    public function getResolver($resolver)
```

Comment:
@johanjanssens you need `$config = array()` as the second parameter here. Otherwise you will get a declaration incompatibility error in PHP

Reply:
True. Good catch. Fixed that.